a10_octavia/controller/worker/tasks/cert_tasks.py

```diff
@@ -24,182 +24,200 @@
+                cert_data = utils.get_cert_data(barbican_client, listener)
+                # Monkey patching template name as listener ID
+                cert_data.template_name = listener.id
+            except Exception as e:
+                LOG.exception("Failed to create SSL certificate: %s", str(e))
+                raise e
+        return cert_data
+
+
 class SSLCertCreate(task.Task):
     """Task to create an SSL certificate"""
     @axapi_client_decorator
-    def execute(self, loadbalancer, listener, vthunder, certificate_type="pem"):
-        barbican_client = BarbicanACLAuth().get_barbican_client(loadbalancer.project_id)
-        cert_data = utils.get_cert_data(barbican_client, listener)
-        try:
-            if self.axapi_client.file.ssl_cert.exists(file=cert_data.cert_filename):
-                self.axapi_client.file.ssl_cert.update(file=cert_data.cert_filename,
-                                                       cert=cert_data.cert_content,
-                                                       size=len(cert_data.cert_content),
-                                                       action="import",
-                                                       certificate_type=certificate_type)
-            else:
-                self.axapi_client.file.ssl_cert.create(file=cert_data.cert_filename,
-                                                       cert=cert_data.cert_content,
-                                                       size=len(cert_data.cert_content),
-                                                       action="import",
-                                                       certificate_type=certificate_type)
-        except Exception as e:
-            LOG.exception("Failed to create SSL certificate: %s", str(e))
-            raise e
-        return cert_data
+    def execute(self, cert_data, vthunder, certificate_type="pem"):
+        if cert_data:
+            try:
+                if self.axapi_client.file.ssl_cert.exists(file=cert_data.cert_filename):
+                    self.axapi_client.file.ssl_cert.update(file=cert_data.cert_filename,
+                                                           cert=cert_data.cert_content,
+                                                           size=len(cert_data.cert_content),
+                                                           action="import",
+                                                           certificate_type=certificate_type)
+                else:
+                    self.axapi_client.file.ssl_cert.create(file=cert_data.cert_filename,
+                                                           cert=cert_data.cert_content,
+                                                           size=len(cert_data.cert_content),
+                                                           action="import",
+                                                           certificate_type=certificate_type)
+            except Exception as e:
```

Comment:
We don't need to do a catch all exceptions here. That will obscure errors. There are only 2 that can be the issue:
1. `ACOSException` from `acos_client`
2. `ConnectTimeout` from `requests library`

We should attempt to catch both and in each case raise the error with the id of the listener/vthunder that is causing the failure.

This modification to exceptions should be complete across the entire refactor when dealing with only the `axapi_client`.

Openstack libraries can raise a larger number of errors which we don't have full visibility of so we can catch-all for them after trying to catch our known errors.